packages/protocol/test/common/metatransactionwallet.ts

```diff
@@ -300,24 +339,80 @@ contract('MetaTransactionWallet', (accounts: string[]) => {
 
       describe('when the caller is the signer', () => {
         describe('when the transactions are executed directly', () => {
-          beforeEach(async () => {
-            await wallet.executeTransactions(
+          describe('', () => {
+            beforeEach(async () => {
+              await wallet.executeTransactions(
+                transactions.map((t) => t.destination),
+                transactions.map((t) => t.value),
+                ensureLeading0x(transactions.map((t) => trimLeading0x(t.data)).join('')),
+                transactions.map((t) => trimLeading0x(t.data).length / 2),
+                { from: signer }
+              )
+            })
+
+            it('should execute the transactions', async () => {
+              assert.equal(await web3.eth.getBalance(transactions[0].destination), value)
+              assert.equal(await web3.eth.getBalance(transactions[2].destination), value)
+              assert.equal(await wallet.signer(), nonSigner)
+            })
+
+            it('should not increment the nonce', async () => {
+              assertEqualBN(await wallet.nonce(), 0)
+            })
+          })
+
+          it('returns the proper values', async () => {
+            const boolTrue: string = '01'
+            const signerAddrReturned: string = trimLeading0x(signer).toLowerCase()
+            const returnValues = await wallet.executeTransactions.call(
               transactions.map((t) => t.destination),
               transactions.map((t) => t.value),
               ensureLeading0x(transactions.map((t) => trimLeading0x(t.data)).join('')),
               transactions.map((t) => trimLeading0x(t.data).length / 2),
               { from: signer }
             )
+
+            assert.equal(
+              returnValues[0],
```

Comment:
can we add some comments about indices 0 and 1 corresponding to the order of `transactions.push`
or alternatively using a normal array init instead of `push`

Reply:
I initialized the array without the push (which seems to make more sense anyway). Not sure if this addresses this comment?

Reply:
yeah that helps
I was also referring to the `returnValues[0]`, `returnValues[1]` "magic indices"

Reply:
ohhhh. I think it's just how web3 handles multiple arguments being returned, since this function returns 2 args. one is the actual data, and one is the lengths array